### PR TITLE
add options to use custom faces

### PR DESCRIPTION
**Pull request by 4hnme**
i'm not sure if this is something that should be added into a simple plugin, but in my setup the tab bar was sticking out too much.
the default look shouldn't change if no new arguments are passed

### CODE REVIEW

**Comment by enricozb** [comment]
Sorry just to confirm you're looking for custom faces instead of the hard-coded faces I use to style the tab bar right?

**Comment by enricozb** [comment]
Whoops thought this was an issue for some reason, I see it's a PR now, I'll take a look thanks :)

File: src/main.rs [hunk 1]
```diff
@@ -21,6 +21,26 @@ pub struct Args {
...
+  #[arg(long, value_name = "FACE")]
+  inactiveface: Option<String>,
+
+  /// Face to use on separators.
+  #[arg(long, value_name = "FACE")]
+  defaultface: Option<String>,
+
+  /// Face to use on a modified tab indicator.
+  #[arg(long, value_name = "FACE")]
+  modifiedface: Option<String>,
```
Comment (enricozb) [inline src/main.rs]: Let's do something like this instead so we don't need `unwrap_or` later:
```suggestion
  /// Separator between tabs.
  #[arg(long, value_name = "FACE", default_value_t="|")]
  separator: String,

  /// Face to use on focused tabs.
  #[arg(long, value_name = "FACE", default_value_t="Prompt")]
  focused_face: String,

  /// Face to use on inactive tabs.
  #[arg(long, value_name = "FACE", default_value_t="LineNumbers")]
  inactive_face: String,

  /// Face to use on separators.
  #[arg(long, value_name = "FACE", default_value_t="Default")]
  default_face: String,

  /// Face to use on a modified tab indicator.
  #[arg(long, value_name = "FACE", default_value_t="DiagnosticError")]
  modified_face: String,
```

File: src/main.rs [hunk 2]
```diff
@@ -39,7 +59,22 @@ pub struct Args {
...
+  }
+  if args.inactiveface == None {
+      args.inactiveface = Some(("LineNumbers").to_string())
+  }
+  if args.defaultface == None {
+      args.defaultface = Some(("Default").to_string())
+  }
+  if args.modifiedface == None {
+      args.modifiedface = Some(("DiagnosticError").to_string())
+  }
```
Comment (enricozb) [inline src/main.rs]: This isn't needed with the above code.

File: src/tabs.rs [hunk 3]
```diff
@@ -18,6 +18,21 @@ pub struct Tabs {
...
+  focusedface: Option<String>,
+
+  /// Face to use on inactive tabs.
+  inactiveface: Option<String>,
+
+  /// Face to use on separators.
+  defaultface: Option<String>,
+
+  /// Face to use on a modified tab indicator.
+  modifiedface: Option<String>,
```
Comment (enricozb) [inline src/tabs.rs]:
```suggestion
  /// Separator between tabs.
  separator: String,

  /// Face to use on focused tabs.
  focused_face: String,

  /// Face to use on inactive tabs.
  inactive_face: String,

  /// Face to use on separators.
  default_face: String,

  /// Face to use on a modified tab indicator.
  modified_face: String,
```

File: src/tabs.rs [hunk 4]
```diff
@@ -48,6 +63,11 @@ impl Tabs {
       modified,
       focused,
       width: args.width,
+      separator: args.separator,
+      focusedface: args.focusedface,
+      inactiveface: args.inactiveface,
+      defaultface: args.defaultface,
+      modifiedface: args.modifiedface,
```
Comment (enricozb) [inline src/tabs.rs]:
```suggestion
      separator: args.separator,
      focused_face: args.focused_face,
      inactive_face: args.inactive_face,
      default_face: args.default_face,
      modified_face: args.modified_face,
```

File: src/tabs.rs [hunk 5]
```diff
@@ -87,13 +107,22 @@ impl Tabs {
...
+            self.inactiveface.as_deref().unwrap_or_default(),
+            self.defaultface.as_deref().unwrap_or_default())
         };
 
         let modified = if self.modified.contains(&i) {
-          " {DiagnosticError}*{Default}".to_string()
+          format!(
+            " {{{}}}*{{{}}}",
+            self.modifiedface.as_deref().unwrap_or_default(),
+            self.defaultface.as_deref().unwrap_or_default())
```
Comment (enricozb) [inline src/tabs.rs]: These and the line below can just use `self.<some>_face` as appropriate.

**Review by enricozb — CHANGES_REQUESTED** [review]

**Comment by enricozb** [comment]
Also for what it's worth I'm currently only using the `client-scope` branch and will probably end up merging that into main. I prefer client-specific buffers instead of sharing them across sessions.

File: src/main.rs (hunk 1 above)
Comment (4hnme) [reply src/main.rs]: my bad. i couldn't figure out how to add default values here

File: src/main.rs [hunk 6]
```diff
@@ -21,6 +21,26 @@ pub struct Args {
...
+  #[arg(long, value_name = "FACE", default_value_t=("LineNumbers").to_string())]
+  inactive_face: String,
+
+  /// Face to use on separators.
+  #[arg(long, value_name = "FACE", default_value_t=("Default").to_string())]
+  default_face: String,
+
+  /// Face to use on a modified tab indicator.
+  #[arg(long, value_name = "FACE", default_value_t=("DiagnosticError").to_string())]
+  modified_face: String,
```
Comment (enricozb) [inline src/main.rs]:
```suggestion
  /// Separator between tabs.
  #[arg(long, value_name = "STR", default_value="|")]
  separator: String,

  /// Face to use on focused tabs.
  #[arg(long, value_name = "FACE", default_value="Prompt")]
  focused_face: String,

  /// Face to use on inactive tabs.
  #[arg(long, value_name = "FACE", default_value="LineNumbers")]
  inactive_face: String,

  /// Face to use on separators.
  #[arg(long, value_name = "FACE", default_value="Default")]
  default_face: String,

  /// Face to use on a modified tab indicator.
  #[arg(long, value_name = "FACE", default_value="DiagnosticError")]
  modified_face: String,
```

File: src/tabs.rs [hunk 7]
```diff
@@ -105,7 +125,7 @@ impl Tabs {
     format!(
       "{}|{}|",
       self.modelinefmt.as_deref().unwrap_or_default(),
-      formatted.join("|")
+      formatted.join(self.separator.as_ref())
```
Comment (enricozb) [inline src/tabs.rs]:
```suggestion
      formatted.join(&self.separator)
```

**Review by enricozb — CHANGES_REQUESTED** [review]
Sorry this can still be a bit shorter (I'll try to commit these myself but I'm not sure if I can edit your branch)

**Comment by enricozb** [comment]
Looks like I was able to commit the changes, thank you for your contribution!

**Pull request merged**